src/relayserver/RelayServer.ts

```diff
@@ -440,50 +439,53 @@ export class RelayServer extends EventEmitter {
     return toBN(await this.contractInteractor.getBalance(this.workerAddress))
   }
 
-  async getShouldRegisterAgain (): Promise<boolean> {
-    const currentBlock = await this.contractInteractor.getBlockNumber()
-    const latestTxBlockNumber = await this._getLatestTxBlockNumber()
+  async getShouldRegisterAgain (currentBlock: number, hubEventsSinceLastScan: EventData[]): Promise<boolean> {
+    const latestTxBlockNumber = await this._getLatestTxBlockNumber(hubEventsSinceLastScan)
     return this.config.registrationBlockRate === 0 ? false : currentBlock - latestTxBlockNumber >= this.config.registrationBlockRate
   }
 
-  async handlePastHubEvents (blockNumber: number): Promise<void> {
+  async handlePastHubEvents (blockNumber: number, hubEventsSinceLastScan: EventData[]): Promise<void> {
+    for (const event of hubEventsSinceLastScan) {
+      switch (event.event) {
+        case TransactionRejectedByPaymaster:
+          log.debug('handle TransactionRejectedByPaymaster event', event)
+          await this._handleTransactionRejectedByPaymasterEvent(blockNumber)
+          break
+      }
+    }
+  }
+
+  async getAllHubEventsSinceLastScan (): Promise<EventData[]> {
     const topics = [address2topic(this.managerAddress)]
     const options = {
       fromBlock: this.lastScannedBlock + 1,
       toBlock: 'latest'
     }
-    const eventNames = [TransactionRejectedByPaymaster]
-    const decodedEvents = await this.contractInteractor.getPastEventsForHub(eventNames, topics, options)
-    for (const dlog of decodedEvents) {
-      switch (dlog.event) {
-        case TransactionRejectedByPaymaster:
-          await this._handleTransactionRejectedByPaymasterEvent(dlog, blockNumber)
-          break
-      }
-    }
+    return await this.contractInteractor.getPastEventsForHub(topics, options)
   }
 
-  async _handleTransactionRejectedByPaymasterEvent (dlog: EventData, blockNumber: number): Promise<void> {
-    log.debug('handle TransactionRejectedByPaymaster event', dlog)
+  async _handleTransactionRejectedByPaymasterEvent (blockNumber: number): Promise<void> {
     this.alerted = true
     this.alertedBlock = blockNumber
     console.error(`Relay entered alerted state. Block number: ${blockNumber}`)
```

Comment:
you removed the reason for being alerted (=the paymaster/tx) even in debug.

Reply:
logged inside the 'switch/case'. It was passed in for no other purpose then being logged.